core/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -202,7 +202,9 @@ public int readVInt() throws IOException {
             return i;
         }
         b = readByte();
-        assert (b & 0x80) == 0;
+        if ((b & 0x80) != 0) {
+            throw new IOException("Invalid vInt " + Integer.toHexString(b) + " & 0x80 = " + Integer.toHexString((b & 0x80)));
```

Comment:
You're correct; I updated it to match the format used in readVLong so it should be more useful